frontend/public/components/resource-dropdown.tsx

```diff
@@ -1,123 +1,50 @@
-
-/*  ------------------------------- NOTE -------------------------------
-
-To avoid circular imports, the keys in this list are manually duplicated in ./resource-pages.tsx !
-
-------------------------------------------------------------------------
-*/
-const resources = [
-  'Alertmanagers',
-  'ClusterServiceVersion-v1s',
-  'Clusters',
-  'ConfigMaps',
-  'CronJobs',
-  'CustomResourceDefinitions',
-  'DaemonSets',
-  'Default',
-  'Deployments',
-  'EtcdClusters',
-  'HorizontalPodAutoscalers',
-  'Ingresses',
-  'Jobs',
-  'Namespaces',
-  'NetworkPolicies',
-  'Nodes',
-  'PersistentVolumeClaims',
-  'PersistentVolumes',
-  'PodVulns',
-  'Pods',
-  'Prometheuses',
-  'ReplicaSets',
-  'ReplicationControllers',
-  'ReportGenerationQuerys',
-  'Reports',
-  'ResourceQuotas',
-  'RoleBindings',
-  'Roles',
-  'Secrets',
-  'ServiceAccounts',
-  'ServiceMonitors',
-  'Services',
-  'StatefulSets',
-  'StorageClasses',
-];
-
-const openshiftResources = [
-  'BuildConfigs',
-  'Builds',
-  'DeploymentConfigs',
-  'ImageStreamTags',
-  'ImageStreams',
-  'Projects',
-  'Routes',
-];
-
-const DropdownItem = ({kind}) => {
-  const [modelRef, kindObj] = allModels().findEntry((v) => v.kind === kind);
-  return <span>
-    <span className="co-type-selector__icon-wrapper">
-      <ResourceIcon kind={modelRef} />
-    </span>
-    {kindObj.labelPlural}
-  </span>;
-};
 
-const ResourceListDropdown_: React.StatelessComponent<ResourceListDropdownProps> = props => {
-  const { selected, onChange, allkinds, openshiftFlag, showAll, className } = props;
-  if (flagPending(openshiftFlag)) {
-    return null;
-  }
-
-  const items: {[s: string]: JSX.Element} = {};
-  const kinds = {};
-  _.each(allModels().toJS(), (ko: K8sKind) => kinds[ko.labelPlural.replace(/ /g, '')] = ko.kind);
-
-  if (showAll) {
-    items.all = <span>
-      <span className="co-type-selector__icon-wrapper">
-        <ResourceIcon kind="All" />
-      </span>All Types</span>;
-  }
-
-  const allResources = openshiftFlag ? _.concat(resources, openshiftResources) : resources;
-  allResources.filter(k => k !== ClusterServiceVersionModel.labelPlural)
-    .sort()
-    .forEach(k => {
-      const kind: string = kinds[k];
-      if (!kind) {
-        return;
-      }
-      const model = allkinds[kind];
-      if (model && model.crd && ![EtcdClusterModel, PrometheusModel, ServiceMonitorModel, AlertmanagerModel].some(m => m.kind === k)) {
-        return;
-      }
-      items[kind] = <DropdownItem kind={kind} />;
-    });
+import { Dropdown, ResourceIcon } from './utils';
+import { K8sKind, K8sResourceKindReference, referenceForModel, apiVersionForReference } from '../module/k8s';
+
+const DropdownItem: React.SFC<DropdownItemProps> = ({model}) => <React.Fragment>
+  <span className="co-type-selector__icon-wrapper">
+    <ResourceIcon kind={model.kind} />
+  </span>
+  {model.kind}&nbsp;<span className="text-muted">({model.apiGroup}/{model.apiVersion})</span>
```

Comment:
Would be nice to only show group/version if model.kind is not unique

Maybe use a smaller font size for group/version as well?